### PR TITLE
max indep set redo

### LINKED ISSUE #3863: Contact Form Submission - Suggestion
Someone submitted the contact form!

**URL**: https://usaco.guide/problems/ys-max-indep-set/solution
**Module**: None
**Topic**: Suggestion
**Message**: 
All these bitwise operation make code unreadable and difficult to understand, please try to be little more verbose.

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

closes #3863
i am a man of my word

### CODE REVIEW

**Review by ryanchou-dev — APPROVED** [review]
immaculate

**Review by envyaims — APPROVED** [review]
among us

File: solutions/gold/ys-MaxIndepSet.mdx [hunk 1]
```diff
@@ -2,216 +2,168 @@
...
-		 * long long d (in binary)
-		 */
-		c = __builtin_ctzll(d);
-		f[i] = popmax(f[i ^ d], f[i & a[c + dt] >> (dt)] | d);
+/** @return a pair with the two things our subsets keep track of */
+std::pair<vector<bool>, vector<ll>> max_cliques(const vector<ll> &adj) {
+	int sz = adj.size();  // just a shorthand
+
+	// This keeps track of whether subsets are cliques themselves
+	vector<bool> valid(1 << sz);
```
Comment (DenialRiver1434) [inline solutions/gold/ys-MaxIndepSet.mdx:79]: would bitset work better than vector<bool> here?

**Review by DenialRiver1434 — CHANGES_REQUESTED** [review]

File: solutions/gold/ys-MaxIndepSet.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/gold/ys-MaxIndepSet.mdx:79]: i mean i'd rather dynamically size it

Comment (DenialRiver1434) [reply solutions/gold/ys-MaxIndepSet.mdx:79]: ah I see

**Review by DenialRiver1434 — APPROVED** [review]

**Pull request merged**